File: apps/agent/agent-api/src/main/java/top/wcpe/beacon/agent/api/AgentIdentity.java
package top.wcpe.beacon.agent.api;

import java.util.Optional;

/**
 * 当前 agent 的身份。
 *
 * <p>{@code group}/{@code zone} 为控制面权威解析回填的值（agent 不声明 zone）；
 * 尚未指派 zone 时 {@link #zone()} 为空。</p>
 */
public final class AgentIdentity {

    private final String namespace;
    private final String serverId;
    private final String role;
    private final String group;
    private final String zone;

    public AgentIdentity(String namespace, String serverId, String role, String group, String zone) {
        this.namespace = namespace;
        this.serverId = serverId;
        this.role = role;
        this.group = group;
        this.zone = zone;
    }

    /** 环境（prod / test）。 */
    public String namespace() {
        return namespace;
    }

    /** 本机唯一身份，环境内唯一。 */
    public String serverId() {
        return serverId;
    }

    /** 角色（bukkit / bungee），由壳层固定。 */
    public String role() {
        return role;
    }

    /** 控制面解析的大区；未解析时为空。 */
    public Optional<String> group() {
        return Optional.ofNullable(group);
    }

    /** 控制面指派的小区；未指派时为空。 */
    public Optional<String> zone() {
        return Optional.ofNullable(zone);
    }
}

File: agent/agent-core/src/main/kotlin/top/wcpe/beacon/agent/core/identity/AgentIdentity.kt
package top.wcpe.beacon.agent.core.identity

/**
 * agent 本机身份（注册请求体的身份与标签部分）。
 *
 * group/zone 不在此声明——zone 由控制面权威指派，group 由控制面解析回填（见注册响应）。
 *
 * @param namespace 环境（prod / test）
 * @param serverId  本机唯一身份，环境内唯一
 * @param role      角色（bukkit / bungee），壳层固定
 * @param groupHint 大区提示（仅未分配 zone 时作兜底 group）
 * @param address   对外可达地址 ip:port
 * @param version   本服业务版本字符串（发现过滤标签，可空）
 * @param capacity  容量（顶层一等字段）
 * @param weight    权重（顶层一等字段）
 * @param metadata  自定义元数据标签（仅 map<string,string>，无 canary）
 */
data class AgentIdentity(
    val namespace: String,
    val serverId: String,
    val role: String,
    val groupHint: String,
    val address: String,
    val version: String,
    val capacity: Int,
    val weight: Int,
    val metadata: Map<String, String>,
) {
    /** 身份是否合法（serverId / namespace 非空）。 */
    fun isValid(): Boolean = serverId.isNotBlank() && namespace.isNotBlank()
}
